src/polyfill.rs

```diff
@@ -178,3 +178,20 @@ macro_rules! slice_as_array_ref_mut {
         }
     }
 }
+
+/// Converts a boolean into a `Result<(), err::Unspecified>` in the
+/// obvious way. By composing this with `try!`, we can compactly
+/// represent `Result`-valued early returns that guard on a condition.
+macro_rules! check {
```

Comment:
This doesn't need to be a macro, does it? I think it can be an inline function.

IMO it would be better to do this as a separate PR that replaced`if x { return Err(...); }` throughout the whole codebase so we can better see the full impact it would have.